cmd/humanlog/genny.go

```diff
@@ -90,18 +99,79 @@ func genny(
 
 	now := start
 	for {
-		if err := emitMessage(out, now, src); err != nil {
-			return err
-		}
-		nextArrival := arrivalRateDist.Rand()
-		nextMsgIn := time.Duration(float64(interval) / nextArrival)
 		select {
 		case <-ctx.Done():
 			return nil
-		case <-time.After(nextMsgIn):
+		default:
+			nextArrival := arrivalRateDist.Rand()
+			nextMsgIn := time.Duration(float64(interval) / nextArrival)
+			time.Sleep(nextMsgIn)
```

Comment:
the prior way of `select`ing on both `ctx.Done()` and `<-time.After` was superior, since it reacted immediately to `ctx` being done. but as of this change, the command will hang until it's done sleeping, before evaluating whether `ctx.Done()` is done.